### PR TITLE
Author profile images frontend

### PR DESCRIPTION
**Pull request by nickgoral2**
This pull request adds author profile images to the bookstore.

The image urls are passed down from back-end via new profile_image_url attribute in the author model.  

Small images of the author are displayed on the book covers and for authors with an empty profile_image_url a placeholder image will be shown instead.

book covers with author images and placeholders

On the author's page the author image is instead displayed at the top of the page in larger size and left off of the book covers.

author page

### CODE REVIEW

**Comment by wcurtis** [comment]
Hey @nickgoral2 - since this is a front-end PR, could you include screenshots or gifs of the update so the reviewers can know what to expect before even setting up locally? Often helps communicate what's changed and speeds up PRs.

**Comment by nickgoral2** [comment]
@wcurtis  No problem, I've added some screenshots.

File: README.md [hunk 1]
```diff
@@ -12,15 +11,22 @@ You will need the following things properly installed on your computer.
...
 
-* `git clone <repository-url>` this repository
+* `git clone https://github.com/nickgoral2/bookstore.git` this repository
 * `cd bookstore`
 * `npm install`
 
 ## Running / Development
 
-* `ember serve`
+* Start back-end server, see instructions at [Bookstore Api](https://github.com/nickgoral2/bookstore-api)
```
Comment (vladucu) [inline README.md]: thanks for the detailed instructions, both here and on the API....really thoughtful ...and clear 👍

File: app/templates/components/book-cover.hbs [hunk 2]
```diff
@@ -1,6 +1,10 @@
 {{!-- app/templates/components/book-cover.hbs --}}
 
 <li {{action "open"}}>
+  {{#if showAuthor}}
+    <img src = "{{ book.author.profile_image_or_placeholder_url}}" alt = "{{book.author.name}}" class = "author-small"/><br>
```
Comment (vladucu) [inline app/templates/components/book-cover.hbs]: any reason for the whitespace around `=`'s?

File: app/templates/components/book-cover.hbs [hunk 3]
```diff
@@ -1,6 +1,10 @@
 {{!-- app/templates/components/book-cover.hbs --}}
 
 <li {{action "open"}}>
+  {{#if showAuthor}}
```
Comment (vladucu) [inline app/templates/components/book-cover.hbs]: this is more of a personal preference, here at Smile, we like declaring properties in the components JS file, with a sane default, which helps to document the component's API

File: app/templates/components/book-cover.hbs (hunk 2 above)
Comment (vladucu) [reply app/templates/components/book-cover.hbs]: also, 👍  for also adding an `alt` tag....should it be of the form `<author-name> image`?

File: app/models/author.js [hunk 4]
```diff
@@ -1,10 +1,20 @@
 import DS from 'ember-data';
 import Ember from 'ember';
 import Publisher from './publisher';
+const { computed } = Ember;
 
 export default Publisher.extend({
+  profile_image_url: DS.attr('string'),
+  profile_image_or_placeholder_url: computed('profile_image_url', function() {
+    if(this.get('profile_image_url')) {
```
Comment (vladucu) [inline app/models/author.js]: missing a space after `if` here, helps with readability

do you think there are any advantages in using one of the Ember computeds for the condition here (isNone, isBlank, isEmpty)?

File: app/templates/components/book-cover.hbs (hunk 3 above)
Comment (nickgoral2) [reply app/templates/components/book-cover.hbs]: Good suggestion, I'll add a sane default.

File: app/models/author.js (hunk 4 above)
Comment (nickgoral2) [reply app/models/author.js]: I'll investigate those options.

**Comment by nickgoral2** [comment]
@vladucu Thanks for the feedback. I'll implement your suggestions later this evening and update the PR.

File: app/models/author.js (hunk 4 above)
Comment (nickgoral2) [reply app/models/author.js]: I will use isBlank.  More readable and also will detect field values containing only whitespace.

File: app/templates/components/book-cover.hbs (hunk 3 above)
Comment (nickgoral2) [reply app/templates/components/book-cover.hbs]: Added default value to book-cover.js

**Comment by nickgoral2** [comment]
@vladucu Hi Vlad I've incorporated your feedback and updated the pull request for your consideration.

File: app/models/author.js (hunk 4 above)
Comment (wcurtis) [reply app/models/author.js]: Yesss - `Ember.isBlank` is one of my favourite things in life 🔑

**Comment by wcurtis** [comment]
🎉

**Comment by wcurtis** [comment]
@nickgoral2 looks like you got the thumbs up from @vladucu - go ahead and 'ship' it! (merge to master) 🚢

**Pull request merged**